source/hwIo.py

```diff
@@ -89,18 +91,24 @@ def waitForRead(self, timeout):
 				timeout -= int((time.time()-curTime)*1000)
 
 	def write(self, data):
+		"""
+		@type data: bytes
+		"""
+		if not isinstance(data, bytes):
+			raise TypeError("Expected argument 'data' to be of type 'bytes'")
 		if _isDebug():
 			log.debug("Write: %r" % data)
 		size = self._writeSize or len(data)
-		buf = ctypes.create_string_buffer(size)
-		buf.raw = data
+		# usage of self._writeSize could result in a size mismatch with data?
+		if size != len(data):
+			log.debugWarning(u"Mismatch in buffer size writing to braille device.")

```

Comment:
I think we only want this if _isDebug()

Reply:
Are you suggesting changing the log level to debug, `len(data)` has a high perf cost, or that this logging is too noisy and we should suppress it unless in a special braille debug mode?

Reply:
hwIo has a _isDebug function that returns True of hwIo specific debugging is True, False otherwise. I think this information should stay under the hwIo debug flag.

Having said that actually, most HID drivers just throw a packet at write that isn't the writeSize, so I'm afraid this will lead to many, many debug warnings. I think it is perfectly valid for drivers to rely on NVDA to send a packet to the device that is of the right size.

It would be good to know what @jcsteh thinks about this, since He's the initial author of hwIo. But my advice would be that if we really want to log this, it should be at debug and behind the hwIo flag. Probably use logHandler.log.isEnabledFor, indeed.